include/executors/executor_sycl.hpp

```diff
@@ -263,7 +262,9 @@ class Executor<SYCL> {
    */
   Executor(cl::sycl::queue q) : q_interface(q){};
 
-  cl::sycl::queue sycl_queue() const { return q_interface.sycl_queue(); }
+  inline Queue_Interface<SYCL> &policy_handler() { return q_interface; }
```

Comment:
Because the queue interface contains the pointer mapper class which deals with allocation and deallocation of buffers and memory. The call to
      implicitly-deleted copy constructor of 'Queue_Interface<SYCL>' error will rise in case of not returning the reference

Reply:
here is the error : 
  'Queue_Interface<SYCL>' is implicitly deleted because field 'pointer_mapper' has a deleted copy
      constructor
  mutable cl::sycl::codeplay::PointerMapper pointer_mapper;
                                            ^
/home/mehdi/soft/sycl-blas/include/queue/pointer_mapper.hpp:310:3: note: 'PointerMapper' has been
      explicitly marked deleted here
  PointerMapper(const PointerMapper&) = delete;

Reply:
used std::shared_ptr to remove the reference